sdk/export/metric/descriptor.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/unit"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+// DescriptorConfig contains configuration for a metric Descriptor
+type DescriptorConfig struct {
+	// Keys are the common keys related to all measurements.
+	Keys []core.Key
+
+	// Description describes the metric in human readable terms.
+	Description string
+
+	// Unit is the determinate quantity used as a standard of all
+	// measurements for the metric.
+	Unit unit.Unit
+
+	// Resource describes the entity for which this metric makes measurements.
+	Resource resource.Resource
+}
+
+// DescriptorOption is the interface that applies the value to a DescriptorConfig option.
+type DescriptorOption interface {
+	// Apply sets the option value of a DescriptorConfig.
+	Apply(*DescriptorConfig)
+}
+
+// WithKeys applies common label keys.
+// Multiple `WithKeys` options accumulate.
+func WithKeys(keys ...core.Key) DescriptorOption {
+	return keysOption(keys)
+}
+
+type keysOption []core.Key
+
+func (k keysOption) Apply(config *DescriptorConfig) {
+	if config == nil {
```

Comment:
Should we even care about `nil` config? `NewDescriptor` never passes `nil` and I don't think these options are going to be used by anyone else.